requirements/requirements.in

```diff
@@ -2,8 +2,8 @@ attrs==18.1.0
 # used to build jsonobject from source
 cython>=0.29.1,<1.0.0
 iso8601==0.1.12
-jsonobject==0.9.4
-jsonobject-couchdbkit==0.9.12
+jsonobject>=0.9.4
+jsonobject-couchdbkit>=0.9.12
```

Comment:
should these be `~=0.9`?

Reply:
I also thought about that when doing this. I think my final thoughts were: any changes to these two libraries are intentionally made for HQ and there's not likely to be any API changes here that need to be thought about in HQ. I think my view of this file is:

* `==` this is either legacy, or we really need to pin this due to a bug or incompatibility in the library or we just don't trust the library to not introduce issues
* `~=` an external library that we want bugfix updates to and are fairly confident they follow semver
* `>=` internal libraries that we release for HQ and will never not want the latest version

That said I haven't actually run that by anyone. @dannyroberts you originally introduced pip-compile, how does that breakdown sound to you?

Reply:
Yeah that makes sense to me. Add to this the fact that (1) the pinned requirements files only get rebuilt when someone does so manually (2) there are separate commands for recompiling with as few changes as possible (`make requirements`) and for recompiling from scratch to get all latest versions (`make upgrade-requirements`), and (3) changes to the pinned requirements files go through code review, and I think there's enough options and human involvement for us to be able to notice and catch upgrades we shouldn't be doing. With that in mind, @emord I think your guidelines there give us a good starting point for sensible "suggested" upgrades when we proactively run `make upgrade-requirements`.

Thanks for articulating it like that, I don't think I ever laid it out so clearly when I originally introduced this.

Reply:
Nice that makes a lot of sense and sounds good. Is it something that should be documented/disseminated somewhere? I remember @dannyroberts gave a nice presentation on pip tools but not sure whether we created any lasting artifacts from that that would be good to have?

Reply:
Hey @czue thanks for asking. When you asked the answer was no, but now thanks to your question the answer is yes :) Check out the new docs at https://confluence.dimagi.com/display/commcarehq/Python+Dependency+Versioning. Also linking them to https://confluence.dimagi.com/display/commcarehq/Home.

Reply:
Nice, thanks! :raised_hands: